### PR TITLE
Return USB HID GET_REPORT requests

### PR DESCRIPTION
**Pull request by jackhumbert**
## Description

This adds the responses to GET_REPORT that the USB HID spec requires. Some of this got removed in #3951, but it's necessary to return the GET_REPORT requests for the keyboard to work in a timely fashion - otherwise it can take up to 20s to start working on Linux (specifically Debian on arm). Other features that have their own interfaces may require the same sort of treatment. My understanding of this is the host queries these interfaces as soon as it gets the USB configuration and descriptors, so if it's in there, it needs to be handled. 

SET_REPORT needs some similar handling for NKRO, but it's optional there, whereas GET_REPORT is required.

Wireshark may be useful in figuring out which systems query which interfaces/reports - that's how I could tell they weren't being responded to.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
`case KEYBOARD_INTERFACE:` has to be hidden behind preprocessors, as it doesn't exist if there is a shared endpoint

**Review by drashna — APPROVED** [review]

File: tmk_core/protocol/chibios/usb_main.c [hunk 1]
```diff
@@ -566,20 +578,38 @@ static bool usb_request_hook_cb(USBDriver *usbp) {
...
+                                    usbSetupTransfer(usbp, (uint8_t *)&mouse_report_sent, sizeof(mouse_report_sent), NULL);
+                                    return TRUE;
+                                    break;                       
+                                }
+#   endif
+#endif /* SHARED_EP_ENABLE */
                             default:
-                                usbSetupTransfer(usbp, NULL, 0, NULL);
+                                universal_report_blank.report_id = usbp->setup[2];
+                                usbSetupTransfer(usbp, (uint8_t *)&universal_report_blank, usbp->setup[6], NULL);
```
Comment (sigprof) [inline tmk_core/protocol/chibios/usb_main.c:612]: Should we guard against an attempt to read the MCU RAM beyond `sizeof(universal_report_blank)` here? Either clamp the size, or just `return FALSE` on a bad value (probably should not care much about the particular kind of error if the host definitely tries to do something wrong).

**Review by sigprof — COMMENTED** [review]
Replying with at least an empty report will probably solve the timeout issues that some people see (although a 100% correct solution would probably be to actually store the last sent report of each type, I'm not sure whether any real HID host actually uses the `GET_REPORT` request outside of the device initialization).  The only remaining issue is handling of some definitely wrong requests (someone may claim that it's a security issue, because technically it could allow someone to read a limited amount of RAM contents that happened to be located after `universal_report_blank`).

**Comment by nblyumberg** [comment]
I had flashed my Preonic v3 with a test build that @tzarc provided and that solved the problem of going into the Preonic becoming non responsive after using layers.

**Review by tzarc — APPROVED** [review]
Let's get this in and iron out the details over the rest of the develop cycle.

**Pull request merged**